### PR TITLE
Bug fix of mutating architectures and hparams simultaneously

### PR DESCRIPTION
**Pull request by ultmaster**
### Description ###

There's a bug when mutating parameters on architectures and hyper-parameters (evaluators) simultaneously. I made a mistake of writing `mutator.name` rather than `mutator.label`, and tests didn't cover this part.

This PR also adds a recursive inspection of evaluator arguments, so that you can do this:

```python
FunctionalEvaluator(
    foo,
    t=Foo(some_argument=ValueChoice([1, 2, 3], label='t')),
    v=ValueChoice([1, 2, 3], label='t') + ValueChoice([10, 20, 30])
)
```

### Checklist ###
  - [x] test case
  - [x] doc

### CODE REVIEW

File: nni/retiarii/experiment/pytorch.py [hunk 1]
```diff
@@ -144,6 +144,10 @@ def preprocess_model(base_model, trainer, applied_mutators, full_ir=True, dummy_
                            'do not use mutators when you use LayerChoice/InputChoice')
     if mutators is not None:
         applied_mutators = mutators
+
+    # Add mutations on evaluators
+    applied_mutators += process_evaluator_mutations(trainer, applied_mutators)
+
```
Comment (QuanluZhang) [inline nni/retiarii/experiment/pytorch.py:150]: I remember some mutators should be put at the end of applied_mutators. could you double check this logic?

Comment (matluster) [reply nni/retiarii/experiment/pytorch.py:150]: That's not relevant to this part. Those belong to model mutators. These are evaluator mutators.

And we don't support sharing choices between model and evaluator.

**Review by liuzhe-lz — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]
